lib/FieldTags/FieldTags.js

```diff
@@ -0,0 +1,97 @@
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { find } from 'lodash';
+import {
+  change,
+  getFormValues,
+} from 'redux-form';
+
+import { FieldMultiSelection } from '../FieldMultiSelection';
+
+const label = <FormattedMessage id="stripes-acq-components.label.tags" />;
+const itemToString = item => item;
+
+const renderTag = ({ filterValue, exactMatch }) => {
+  if (exactMatch || !filterValue) {
+    return null;
+  } else {
+    return (
+      <FormattedMessage
+        id="stripes-acq-components.addTagFor"
+        values={{ filterValue }}
+      />
+    );
+  }
+};
+
+const FieldTags = ({
+  allTags,
+  formName,
+  name,
+  onAdd,
+  store,
+}) => {
+  const addTag = useCallback(
+    ({ inputValue }) => {
+      const tag = inputValue.replace(/\s|\|/g, '').toLowerCase();
+      const formValues = getFormValues(formName)(store.getState());
+      const updatedTags = formValues.tags.tagList.concat(tag).filter(Boolean);
+
+      if (tag) {
+        store.dispatch(change(formName, name, updatedTags));
+        onAdd(tag);
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [formName, onAdd],
+  );
+
+  const addAction = { onSelect: addTag, render: renderTag };
+  const actions = [addAction];
+
+  const dataOptions = allTags.map(tag => tag.label.toLowerCase()).sort();
+
+  const formatter = ({ option }) => {
+    const item = find(allTags, { label: option }) || option;
+
+    if (!item) return option;
+
+    return item.label;
+  };
+
+  const filterItems = (filterText, list) => {
+    const renderedItems = filterText
+      ? list.filter(tag => tag.toLowerCase().includes(filterText.toLowerCase()))
```

Comment:
please look at https://github.com/folio-org/stripes-acq-components/blob/master/lib/utils/filterSelectValues.js

Reply:
Reused it. In my case `dataOptions` it's an array of strings

Reply:
it's still a common code. I think it might be useful to leverage this into a separate common fn, something like `filterArrayValues`

Reply:
Done